app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{"
+                f"Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}"
+                f"}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, other: Herbivore) -> str:
+        if other.hidden:
+            return f"{self.name} cannot bite hidden {other.name}"
+        if isinstance(other, Herbivore):
+            other.health -= 50
+            if other.health <= 0:
+                other.alive.remove(other)
```

Comment:
```suggestion
        if isinstance(other, Herbivore):
            if other.hidden:
                return f"{self.name} cannot bite hidden {other.name}"
            
            other.health -= 50
            if other.health <= 0:
                Animal.alive.remove(other)
```

First, you'd better check that other is instance of the Herbivore